packages/experiments/src/components/SelectedItemsList/SelectedItemsList.tsx

```diff
@@ -1,131 +1,221 @@
+  }, [controlledRef]);
 
   // For usage as a controlled component with a ref
   React.useImperativeHandle(
-    ref,
+    props.componentRef,
     (): ISelectedItemsList<TItem> => ({
-      items,
+      items: selectedItems,
+      addItems: appendSelectedItems,
+      removeItems: removeSelectedItems,
       itemsInSelection,
-      addItems,
       unselectAll,
-      removeItems
+      copyItemsInSelectionToClipboard
     }),
-    [items, addItems]
+    [selectedItems, itemsInSelection, appendSelectedItems, unselectAll, removeSelectedItems, copyItemsInSelectionToClipboard]
   );
 
-  const onRemoveItemCallbacks = React.useMemo(
-    () =>
-      // create callbacks ahead of time with memo.
-      // (hooks have to be called in the same order)
-      items.map((item: TItem) => () => removeItems([item])),
-    [items]
-  );
-
-  const SelectedItem = props.onRenderItem;
   return (
-    <>
-      {renderedItems.map((item: TItem, index: number) => (
-        <SelectedItem
-          item={item}
-          index={index}
-          key={item.key !== undefined ? item.key : index}
-          selected={selectedIndices.indexOf(index) !== -1}
-          removeButtonAriaLabel={props.removeButtonAriaLabel}
-          onRemoveItem={onRemoveItemCallbacks[index]}
-          onItemChange={replaceItem}
-          onCopyItem={copyItemsInSelection}
-        />
-      ))}
-    </>
+    <ControlledSelectedItemsList<TItem>
+      componentRef={controlledRef}
+      // passthrough props
+      getItemCopyText={props.getItemCopyText}
+      onRenderItem={props.onRenderItem}
+      removeButtonAriaLabel={props.removeButtonAriaLabel}
+      canRemoveItem={props.canRemoveItem}
+      // props that are internal to the uncontrolled component
+      selection={selection}
+      selectedItems={selectedItems}
+      onItemsRemoved={removeSelectedItems}
+      onItemChange={replaceSelectedItem}
+    />
   );
 };
-
-// Typescript only respects unifying a generic type with a generic const _function_ of the same name for function types.
-// In order to satisfy the type checker, here we lie about the type of the const so that it is still a generic function.
-export type SelectedItemsList<TItem extends BaseSelectedItem> = React.Component<ISelectedItemsListProps<TItem>>;
-export const SelectedItemsList = React.forwardRef(_SelectedItemsList) as (typeof _SelectedItemsList);
+export type UncontrolledSelectedItemsList<TItem extends BaseSelectedItem> = (
+  props: IUncontrolledSelectedItemListProps<TItem>
+) => React.ReactElement;
+(UncontrolledSelectedItemsList as any).displayName = 'UncontrolledSelectedItemsList';
+
+const isControlledSelectedItemList = <T extends BaseSelectedItem>(
+  props: IControlledSelectedItemListProps<T> | IUncontrolledSelectedItemListProps<T>
+): props is IControlledSelectedItemListProps<T> => (props as any).selectedItems !== undefined;
+
+export const SelectedItemsList = <TItem extends BaseSelectedItem>(props: ISelectedItemsListProps<TItem>) => {
```

Comment:
Given the importance of the `selectedItems` prop, maybe include a longer description of what will happen if you pass that prop vs. not in the `SelectedItemsList.types.ts` docstrings. Something like "If you pass this prop, you are taking full responsibility for managing the state of the selection and all selection-changing callbacks"